React/Base/RCTComponentEvent.m

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTComponentEvent.h"
+
+#import "RCTAssert.h"
+
+@implementation RCTComponentEvent
+{
+  NSArray *_arguments;
+}
+
+@synthesize eventName = _eventName;
+@synthesize viewTag = _viewTag;
+
+- (instancetype)initWithName:(NSString *)name body:(NSDictionary *)body
+{
+  if (self = [super init]) {
+    NSNumber *target = body[@"target"];
+    name = RCTNormalizeInputEventName(name);
+    
```

Comment:
`#ifdef` ?

Reply:
Yea that would be better (just copied that code :))